DependencyInjectionContainer/Service.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Reflection;
+
+namespace DependencyInjectionContainer;
+using Enums;
+
+internal sealed class Service
+{
+    internal Service(Type interfaceType, Type implementationType, ServiceLifetime lifetime, Func<DIContainer, object> implementationFactory = null)
```

Comment:
consider exposing the most specific constructors which delegates parameters to private one which accept all possible parameters